### PR TITLE
New App: Vercel dashboard

### PR DESCRIPTION
**Pull request by croossin**
This gives ability for developers to see the latest deployment information from their Vercel dashboard.

It includes:
- Commit message
- Project name
- Author name
- Success/Failure
- Time it was created

### CODE REVIEW

File: apps/verceldashboard/vercel_dashboard.star [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+def main(config):
+    apikey = config.str(CONFIG_API_KEY)
+    showIn24hr = config.bool(CONFIG_CLOCK_FORMAT, True)
+
+    # Figure out how to show if no api key
+    if apikey == None:
+        fail("No API Key provided")
+
+    cached_data = cache.get(CACHE_KEY)
```
Comment (matslina) [inline apps/verceldashboard/vercel_dashboard.star:32]: Cache is scoped to your app, but not to individual users. This record will hold whatever happened to get written to cache last time. You likely need to incorporate something like the API key into the cache key.

**Review by matslina — CHANGES_REQUESTED** [review]

File: apps/verceldashboard/vercel_dashboard.star (hunk 1 above)
Comment (croossin) [reply apps/verceldashboard/vercel_dashboard.star:32]: Oh very interesting, I did not realize that when reading the docs - must have overlooked. Will fix that.

File: apps/verceldashboard/vercel_dashboard.star [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = CONFIG_API_KEY,
+                name = "API Key",
+                desc = "Your Vercel API key generated via the Vercel dashboard",
+                icon = "key",
+            ),
+            schema.Dropdown(
```
[the remark this answers is not in the document]
Comment (croossin) [reply apps/verceldashboard/vercel_dashboard.star:152]: Also was not aware of `$tz`. I guess I could use that as default, but I do like the idea of having the option to change as some developers may enjoy seeing it in another timezone - UTC as an example.

File: apps/verceldashboard/vercel_dashboard.star [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+CONFIG_CLOCK_FORMAT = "time-format"
+CONFIG_TIMEZONE = "time-zone"
+
+def main(config):
+    apikey = config.str(CONFIG_API_KEY)
+    showIn24hr = config.bool(CONFIG_CLOCK_FORMAT, True)
+
+    # Figure out how to show if no api key
+    if apikey == None:
+        fail("No API Key provided")
```
[the remark this answers is not in the document]
Comment (croossin) [reply apps/verceldashboard/vercel_dashboard.star:30]: Certainly. I think my comment to myself even suggests that but must have totally slipped my mind. Will add that as well.

File: apps/verceldashboard/vercel_dashboard.star (hunk 1 above)
Comment (croossin) [reply apps/verceldashboard/vercel_dashboard.star:32]: @matslina Do we need to encode/decode the API key if we are going to use that as part of the cache key? Or is the security in place that the cache key is not exposed anywhere?

Comment (matslina) [reply apps/verceldashboard/vercel_dashboard.star:34]: The cache key will not be visible to other apps, so using it directly should be secure. That said, there's nothing wrong with computing a hash of it and using that instead.

Comment (croossin) [reply apps/verceldashboard/vercel_dashboard.star]: Makes sense - I have went ahead and updated it with a hashed version of the api key!

**Comment by croossin** [comment]
@matslina I believe I took care of all the requests - let me know if there is anything else you would recommend changing. Thanks again for the review!

**Comment by croossin** [comment]
@matslina It seems that the Actions are backed up and not being queued properly? Just a heads up.

**Pull request merged**